test/util/util.go

```diff
@@ -245,35 +245,41 @@ deployWatch:
 	}
 }
 
-func EventSourcePodLogContains(ctx context.Context, kubeClient kubernetes.Interface, namespace, eventSourceName, regex string, timeout time.Duration) (bool, error) {
+func EventSourcePodLogContains(ctx context.Context, kubeClient kubernetes.Interface, namespace, eventSourceName, regex string, timeout time.Duration, countOpt *int) (bool, error) {
 	labelSelector := fmt.Sprintf("controller=eventsource-controller,eventsource-name=%s", eventSourceName)
 	podList, err := kubeClient.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: "status.phase=Running"})
 	if err != nil {
 		return false, fmt.Errorf("error getting event source pod name: %w", err)
 	}
 
-	return PodsLogContains(ctx, kubeClient, namespace, regex, podList, timeout), nil
+	return PodsLogContains(ctx, kubeClient, namespace, regex, podList, timeout, countOpt), nil
 }
 
-func SensorPodLogContains(ctx context.Context, kubeClient kubernetes.Interface, namespace, sensorName, regex string, timeout time.Duration) (bool, error) {
+func SensorPodLogContains(ctx context.Context, kubeClient kubernetes.Interface, namespace, sensorName, regex string, timeout time.Duration, countOpt *int) (bool, error) {
 	labelSelector := fmt.Sprintf("controller=sensor-controller,sensor-name=%s", sensorName)
 	podList, err := kubeClient.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: "status.phase=Running"})
 	if err != nil {
 		return false, fmt.Errorf("error getting sensor pod name: %w", err)
 	}
 
-	return PodsLogContains(ctx, kubeClient, namespace, regex, podList, timeout), nil
+	return PodsLogContains(ctx, kubeClient, namespace, regex, podList, timeout, countOpt), nil
 }
 
-func PodsLogContains(ctx context.Context, kubeClient kubernetes.Interface, namespace, regex string, podList *corev1.PodList, timeout time.Duration) bool {
+func PodsLogContains(ctx context.Context, kubeClient kubernetes.Interface, namespace, regex string, podList *corev1.PodList, timeout time.Duration, countOpt *int) bool {
```

Comment:
A better approach would be using options like below, and the function signature accepts `...PodLogCheckOption` so that all the invokers will not need to add `nil` as a parameter.

type podLogCheckOptions struct {
  timeout time.Duration
  count int
}

func defaultPodLogCheckOptions() *podLogCheckOptions {
  return &podLogCheckOptions {
    timeout: xxxx,
    count: -1,
  }
}

type PodLogCheckOption func(*podLogCheckOptions)

func WithTimeout(t time.Duration) PodLogCheckOption {
  return func(o *podLogCheckOptions) {
    o.timeout = t
  }
}

func WithCount(c int) PodLogCheckOption {
  return func(o *podLogCheckOptions) {
    o.count = c
  }
}

Reply:
encapsulating timeout and count into a struct sounds good. So, you're recommending a variadic parameter so that the invoker doesn't need to pass anything in? One thing I'm wondering about is that a variadic parameter also implies that you can have more than 1, which would be odd in this case. 

Do we sometimes use variadic parameters but only look at the first one specified?

Reply:
The way it handles it is like below:

unc PodsLogContains(ctx context.Context, kubeClient kubernetes.Interface, namespace, regex string, podList *corev1.PodList, opts ...PodLogCheckOption) bool {
  o := defaultPodLogCheckOptions()
  for _, opt := range opts {
    if opt != nil {
      opt(o)
    }

    // then you can use `o.timeout` and `o.count`.
  }




}

Reply:
I see